`Surface_mesh_parameterization` : call to `is_one_to_one_mapping` is ambiguous

## Issue Details

`LSCM_parameterizer_3::is_one_to_one_mapping` (or `ARAP_parameterizer_3::is_one_to_one_mapping`) call returns an ambiguous call error between those two methods in file `Surface_mesh_parameterization/internal/validity.h`. 

```c++
template <typename TriangleMesh,
          typename Faces_Container,
          typename VertexUVMap>
bool is_one_to_one_mapping(const TriangleMesh& mesh,
                           const Faces_Container& faces,
                           const VertexUVMap uvmap)
```
and 
```c++
template <typename TriangleMesh,
          typename VertexUVMap>
bool is_one_to_one_mapping(const TriangleMesh& mesh,
                           typename boost::graph_traits<TriangleMesh>::halfedge_descriptor bhd,
                           const VertexUVMap uvmap)
```
Here is the error :
```
In file included from /path/to/CGAL/Surface_mesh_parameterization/ARAP_parameterizer_3.h:26:
/path/to/CGAL/Surface_mesh_parameterization/LSCM_parameterizer_3.h:164:12: error: call to 'is_one_to_one_mapping' is ambiguous
    return internal::is_one_to_one_mapping(mesh, bhd, uvmap);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/path/to/myTest.cpp:271:46: note: in instantiation of function template specialization 'CGAL::Surface_mesh_parameterization::LSCM_parameterizer_3<CGAL::Seam_mesh<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double>>>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double>>>::Property_map<CGAL::SM_Edge_index, bool>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double>>>::Property_map<CGAL::SM_Vertex_index, bool>>, CGAL::Surface_mesh_parameterization::Two_vertices_parameterizer_3<CGAL::Seam_mesh<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double>>>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double>>>::Property_map<CGAL::SM_Edge_index, bool>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double>>>::Property_map<CGAL::SM_Vertex_index, bool>>>>::is_one_to_one_mapping<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double>>>::Property_map<CGAL::SM_Halfedge_index, CGAL::Point_2<CGAL::Simple_cartesian<double>>>>' requested here
        std::cout << "one-to-one ?" << parametrizer.is_one_to_one_mapping(mesh, bhd, uv_pm) << std::endl;
                                                    ^
/path/to/CGAL/Surface_mesh_parameterization/internal/validity.h:241:6: note: candidate function [with TriangleMesh = CGAL::Seam_mesh<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double>>>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double>>>::Property_map<CGAL::SM_Edge_index, bool>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double>>>::Property_map<CGAL::SM_Vertex_index, bool>>, Faces_Container = CGAL::Seam_mesh_halfedge_descriptor<CGAL::SM_Halfedge_index>, VertexUVMap = CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double>>>::Property_map<CGAL::SM_Halfedge_index, CGAL::Point_2<CGAL::Simple_cartesian<double>>>]
bool is_one_to_one_mapping(const TriangleMesh& mesh,
     ^
/path/to/CGAL/Surface_mesh_parameterization/internal/validity.h:294:6: note: candidate function [with TriangleMesh = CGAL::Seam_mesh<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double>>>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double>>>::Property_map<CGAL::SM_Edge_index, bool>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double>>>::Property_map<CGAL::SM_Vertex_index, bool>>, VertexUVMap = CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double>>>::Property_map<CGAL::SM_Halfedge_index, CGAL::Point_2<CGAL::Simple_cartesian<double>>>]
bool is_one_to_one_mapping(const TriangleMesh& mesh,
     ^
1 error generated.
```

## Source Code

Code directly adapted from `examples/Surface_mesh_parameterization/lscm.cpp`, only call to `is_one_to_one_mapping` was added.

```c++
typedef CGAL::Simple_cartesian<double>      Kernel;
typedef Kernel::Point_2                     Point_2;
typedef Kernel::Point_3                     Point_3;
typedef CGAL::Surface_mesh<Kernel::Point_3> SurfaceMesh;
typedef boost::graph_traits<SurfaceMesh>::edge_descriptor SM_edge_descriptor;
typedef boost::graph_traits<SurfaceMesh>::halfedge_descriptor SM_halfedge_descriptor;
typedef boost::graph_traits<SurfaceMesh>::vertex_descriptor SM_vertex_descriptor;
typedef SurfaceMesh::Property_map<SM_halfedge_descriptor, Point_2> UV_pmap;
typedef SurfaceMesh::Property_map<SM_edge_descriptor, bool> Seam_edge_pmap;
typedef SurfaceMesh::Property_map<SM_vertex_descriptor, bool> Seam_vertex_pmap;
typedef CGAL::Seam_mesh<SurfaceMesh, Seam_edge_pmap, Seam_vertex_pmap> Mesh;
typedef boost::graph_traits<Mesh>::vertex_descriptor vertex_descriptor;
typedef boost::graph_traits<Mesh>::halfedge_descriptor halfedge_descriptor;
typedef boost::graph_traits<Mesh>::face_descriptor face_descriptor;

namespace SMP = CGAL::Surface_mesh_parameterization;

void test()
{
	const char* filename = "data/lion.off";
	SurfaceMesh sm;
	if(!CGAL::IO::read_polygon_mesh(filename, sm))
	{
		std::cerr << "Invalid input file." << std::endl;
	}
	else
	{
		// Two property maps to store the seam edges and vertices
		Seam_edge_pmap seam_edge_pm = sm.add_property_map<SM_edge_descriptor, bool>("e:on_seam", false).first;
		Seam_vertex_pmap seam_vertex_pm = sm.add_property_map<SM_vertex_descriptor, bool>("v:on_seam", false).first;
		const char* selection_filename = "data/lion.selection.txt";
		// Read the constraints on the border
		std::ifstream in(filename);
		std::string vertices;
		std::getline(in, vertices);
		std::istringstream iss(vertices);

		Mesh mesh(sm, seam_edge_pm, seam_vertex_pm);
		SM_halfedge_descriptor smhd = mesh.add_seams(selection_filename);
		if(smhd == SM_halfedge_descriptor() ) {
		  std::cerr << "Warning: No seams in input" << std::endl;
		}
		// The 2D points of the uv parametrisation will be written into this map
		// Note that this is a halfedge property map, and that uv values
		// are only stored for the canonical halfedges representing a vertex
		UV_pmap uv_pm = sm.add_property_map<SM_halfedge_descriptor, Point_2>("h:uv").first;
		// A halfedge on the (possibly virtual) border
		halfedge_descriptor bhd = CGAL::Polygon_mesh_processing::longest_border(mesh, CGAL::Polygon_mesh_processing::parameters::all_default()).first;
		typedef SMP::Two_vertices_parameterizer_3<Mesh>                Border_parameterizer;
		typedef SMP::LSCM_parameterizer_3<Mesh, Border_parameterizer>  Parameterizer;

		Parameterizer parametrizer = Parameterizer();
		SMP::parameterize(mesh, parametrizer, bhd, uv_pm);

		// ////////////////////////////////////////////////////////////////////
		// this line does not compile : ambiguous call
		std::cout << "one-to-one ?" << parametrizer.is_one_to_one_mapping(mesh, bhd, uv_pm) << std::endl;
		// ////////////////////////////////////////////////////////////////////

		std::ofstream out("/tmp/result.off");
		SMP::IO::output_uvmap_to_off(mesh, bhd, uv_pm, out);
	}
}
```

## Environment

* Operating system : Debian 10
* Compiler: clang 12
* CGAL version: 5.3

--- a/Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/internal/validity.h
+++ b/Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/internal/validity.h
@@ -26,7 +26,8 @@
 #include <CGAL/Polygon_mesh_processing/connected_components.h>
 
 #include <boost/iterator/function_output_iterator.hpp>
-
+#include <boost/utility/enable_if.hpp>
+#include <boost/range/has_range_iterator.hpp>
 #include <vector>
 
 namespace CGAL {
@@ -240,7 +241,10 @@ template <typename TriangleMesh,
           typename VertexUVMap>
 bool is_one_to_one_mapping(const TriangleMesh& mesh,
                            const Faces_Container& faces,
-                           const VertexUVMap uvmap)
+                           const VertexUVMap uvmap,
+                           typename boost::enable_if<
+                              boost::has_range_iterator<Faces_Container>
+                           >::type* = nullptr)
 {
   typedef typename boost::graph_traits<TriangleMesh>::vertex_descriptor    vertex_descriptor;
   typedef typename boost::graph_traits<TriangleMesh>::halfedge_descriptor  halfedge_descriptor;

